tests/foreman/cli/test_fact.py

```diff
@@ -61,3 +61,56 @@ def test_negative_list_by_name(module_target_sat):
     assert (
         module_target_sat.cli.Fact().list(options={'search': f'fact={gen_string("alpha")}'}) == []
     )
+
+
+def test_positive_custom_facts(module_target_sat):
+    """Create custom facts and verify they are updated on Satellite
+
+    :id: ae0b5574-cc8b-4f0c-ba7b-c6f480c08e06
+
+    :steps:
+        1. Create few custom facts
+        2. Verify on Satellite
+
+    :expectedresults: Custom facts are created and uploaded successfully.
+    """
+    ptable = module_target_sat.cli_factory.make_partition_table()
+    os = module_target_sat.cli_factory.make_os()
+    module_target_sat.cli.PartitionTable.add_operating_system(
+        {'name': ptable['name'], 'operatingsystem': os['title']}
+    )
+    ptable = module_target_sat.cli.PartitionTable.info({'name': ptable['name']})
+    assert os['title'] in ptable['operating-systems']
+
+
+@pytest.mark.parametrize(
+    'fact', ['uptime', 'os::family', 'uptime_seconds', 'memorysize', 'ipaddress']
+)
+def test_positive_verify_imported_values(fact, module_target_sat):
+    """Verify imported values
+
+    :id: 64f19639-81d2-47e2-a10d-8456b07b4828
+
+    :steps:
+        1. upload all fact values on Satellite
+        2. Verify on Satellite
+
+    :expectedresults: All facts values are verified.
+    """
+    facts = module_target_sat.cli.Fact().list(options={'search': f'fact={fact}'})
+    assert facts[0]['fact'] == fact
+
+
+def test_positive_facts_cleanup(target_sat):
+    """Verify facts are cleaned up
+
+    :id: 428cb2db-3edf-4d71-954f-c1903896332e
+
+    :steps:
+        1. Clean some facts for host
+        2. Verify on Satellite
+
+    :expectedresults: All orphaned facts values are cleaned up.
+    """
+    result = target_sat.execute('foreman-rake facts:clean')
+    assert result.status == 0
```

Comment:
Are there always facts to clean on a default satellite? Or do we need to add some and verify they are removed? Right now it seems that we are only asserting the rake command runs without any errors.

Reply:
Would like to add something from my end, I checked hammer facts command, didn't see any option to clean, but yes +1 to add some more verification steps.

Reply:
@Griffin-Sullivan Only orphaned facts will be cleaned. I have asserted the message for more verification.

Reply:
@vijaysawant Yes, you are correct. There are no hammer command to clean the orphaned facts so I have used this one.